### PR TITLE
Dependency updates

### PR DESCRIPTION
**Pull request by shiftgeist**
- Update hexo (and subcomponents) to 4.2.0.
- Package `hoek` moved to `@hapi/hoek` and updated to 9.0.3
- Replaced `request` (https://github.com/request/request/issues/3142) with "node-fetch".
- Adjusted download function to work with node-fetch.
- Adjust `.npmrc` to 10.19.0 since node 8 is no longer in maintenance mode ([nodejs/releases](https://nodejs.org/en/about/releases/)) (hexo 4.1.0 [announced to droped support](https://github.com/hexojs/hexo/releases/tag/4.1.0) for node 8)

### CODE REVIEW

File: pre-deploy.js [hunk 1]
```diff
@@ -49,26 +49,27 @@ Promise.all([
...
-      if (res.statusCode != 200) {
-        return reject(
-          `unexpected response code when downloading from unpkg: ${res.statusCode}` +
-          `\n${body.toString()}`
-        )
-      }
-      fs.writeFile(`themes/vue/source/js/${file}`, body, err => {
-        if (err) return reject(err)
-        zlib.gzip(body, (err, zipped) => {
+    fetch(`http://unpkg.com/vue@${version}/dist/${file}`)
```
Comment (phanan) [inline pre-deploy.js]:
```suggestion
    fetch(`https://unpkg.com/vue@${version}/dist/${file}`)
```

Might as well use HTTPS now.

File: pre-deploy.js [hunk 2]
```diff
@@ -49,26 +49,27 @@ Promise.all([
...
+      .then(function(res) {
```
Comment (phanan) [inline pre-deploy.js]:
```suggestion
      .then(res => {
```

File: pre-deploy.js [hunk 3]
```diff
@@ -49,26 +49,27 @@ Promise.all([
...
+        if (res.status != 200) {
```
Comment (phanan) [inline pre-deploy.js]:
```suggestion
        if (res.status !== 200) {
```

File: pre-deploy.js [hunk 4]
```diff
@@ -49,26 +49,27 @@ Promise.all([
...
+          return reject(
+            `unexpected response code when downloading from unpkg: ${res.status}` +
+            `\n${res.text()}`
+          )
+        }
+        return res.text()
+      })
+      .then(function(body) {
```
Comment (phanan) [inline pre-deploy.js]:
```suggestion
      .then(body => {
```

File: pre-deploy.js [hunk 5]
```diff
@@ -49,26 +49,27 @@ Promise.all([
...
+          zlib.gzip(body, (err, zipped) => {
+            if (err) return reject(err)
+            resolve((zipped.length / 1024).toFixed(2))
+          })
         })
       })
-    })
+      .catch(function(err) {
+        return reject(err.statusText)
+      })
```
Comment (phanan) [inline pre-deploy.js]:
```suggestion
      .catch(err => reject(err.statusText))
```

File: pre-deploy.js [hunk 6]
```diff
@@ -1,8 +1,8 @@
-// udpate to latest built files of Vue
+// update to latest built files of Vue
```
Comment (phanan) [inline pre-deploy.js]: Ha!

**Comment by shiftgeist** [comment]
Thanks @phanan for checking the pr, i'll look into it :)

**Comment by shiftgeist** [comment]
The API menu is still not working and I don't know why.

**Comment by phanan** [comment]
I don't have too much time to look at this now but apparently some of the scripts (including the critical `common.js`) are not being loaded. Maybe you want to investigate into that direction?

**Comment by shiftgeist** [comment]
This checked out to be the right direction: All script tags cannot be found in the vdom.

**Comment by shiftgeist** [comment]
[v2/api](https://deploy-preview-2518--vuejs.netlify.com/v2/api/) alive and well :)

**Comment by phanan** [comment]
Cool! On the [Style Guide] page though, there's a bunch of `%raw%` and `%endraw% thingies, you want to take a thorough look through everything?

**Comment by shiftgeist** [comment]
I'll take a look.

**Comment by shiftgeist** [comment]
Turns out the `% raw %` and `% endraw %` were not needed.

Also cleaned up the markdown markup of the style-guide.

**Comment by phanan** [comment]
Thanks, but if you notice, the style has been broken. Here's your version:

image

And here's the orignal:

image

Also, as I said above, it's not just the Style Guide, but other pages like Introduction as well.

**Comment by shiftgeist** [comment]
Had to adjust come styles, because the HTML markup of highlighted code blocks changed:

Before:

image


Now:

image

My style adjustments: [page.styl](https://github.com/vuejs/vuejs.org/pull/2518/files#diff-9e5256185a403f3eae79a0b15fabb496)

**Comment by shiftgeist** [comment]
*ping* Is there anything else that is missing?

**Comment by shiftgeist** [comment]
@phanan Any status updates?

**Comment by phanan** [comment]
Sorry for losing track. The style is still broken for me: 

image

However, before you continue: We've decided to pause changes made to v2 docs to ensure a smooth migrating to v3 (in which most of the changes we're making here will become less relevant anyway). As a result, I'll have to go ahead and close the PR. Thanks for the effort, and hope to see your next contributions!

**Pull request closed without merging**